### ISSUE TITLE
🐛 SQL errors on INSERTs to char_effects

### ISSUE THREAD

**Issue by CatsEyeXI** [DerpyProjectGroup/topaz#2265]
- [x] I have paid attention to this example and will edit again if need be to not break the formatting, or I will be ignored
- [x] I have searched existing [issues](https://github.com/LandSandBoat/server/issues) to see if the issue has already been opened, and I have checked the commit log to see if the issue has been resolved since my server was last updated
- [x] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md)

## Branch affected by issue 

`base`

## Steps to reproduce
Is there any explanation for this, or just something I have going on?

```
PS C:\catseyexi_new\log> findstr char_effect .\map-server.log
[07/08/22 13:39:18:908][map][error] Query: INSERT INTO char_effects (charid, effectid, icon, power, tick, duration, subid, subpower, tier, flags, timestamp) VALUES(548,321,321,12,0,233,17241986,4,73,131368,1657301891); (SqlConnection::QueryStr:364)
[07/08/22 13:39:18:908][map][error] Query: INSERT INTO char_effects (charid, effectid, icon, power, tick, duration, subid, subpower, tier, flags, timestamp) VALUES(548,310,310,12,0,235,17241986,5,288,131368,1657301894); (SqlConnection::QueryStr:364)
[07/08/22 16:03:35:244][map][error] Query: INSERT INTO char_effects (charid, effectid, icon, power, tick, duration, subid, subpower, tier, flags, timestamp) VALUES(548,317,317,25,0,132,17586054,4,62,131368,1657310447); (SqlConnection::QueryStr:364)
[07/08/22 16:03:35:245][map][error] Query: INSERT INTO char_effects (charid, effectid, icon, power, tick, duration, subid, subpower, tier, flags, timestamp) VALUES(548,310,310,4,0,120,17586054,4,288,131368,1657310435); (SqlConnection::QueryStr:364)
[07/09/22 18:22:32:303][map][error] Query: INSERT INTO char_effects (charid, effectid, icon, power, tick, duration, subid, subpower, tier, flags, timestamp) VALUES(1379,310,310,3,0,226,17430404,3,288,131368,1657405279); (SqlConnection::QueryStr:364)
[07/09/22 18:22:32:303][map][error] Query: INSERT INTO char_effects (charid, effectid, icon, power, tick, duration, subid, subpower, tier, flags, timestamp) VALUES(1379,321,321,8,0,207,17430404,1,73,131368,1657405260); (SqlConnection::QueryStr:364)
[07/10/22 10:18:24:113][map][error] Query: INSERT INTO char_effects (charid, effectid, icon, power, tick, duration, subid, subpower, tier, flags, timestamp) VALUES(918,324,324,2,0,148,17651592,6,369,131368,1657462552); (SqlConnection::QueryStr:364)
[07/10/22 10:18:24:113][map][error] Query: INSERT INTO char_effects (charid, effectid, icon, power, tick, duration, subid, subpower, tier, flags, timestamp) VALUES(918,314,314,3,0,151,17651592,2,30,131368,1657462555); (SqlConnection::QueryStr:364)
[07/10/22 12:46:55:802][map][error] Query: INSERT INTO char_effects (charid, effectid, icon, power, tick, duration, subid, subpower, tier, flags, timestamp) VALUES(1701,321,321,11,0,231,17201035,4,73,131368,1657471547); (SqlConnection::QueryStr:364)
[07/10/22 12:46:55:802][map][error] Query: INSERT INTO char_effects (charid, effectid, icon, power, tick, duration, subid, subpower, tier, flags, timestamp) VALUES(1701,310,310,3,0,234,17201035,3,288,131368,1657471549); (SqlConnection::QueryStr:364)
[07/10/22 12:47:00:800][map][error] Query: INSERT INTO char_effects (charid, effectid, icon, power, tick, duration, subid, subpower, tier, flags, timestamp) VALUES(1701,321,321,11,0,226,17201035,4,73,131368,1657471547); (SqlConnection::QueryStr:364)
[07/10/22 12:47:00:800][map][error] Query: INSERT INTO char_effects (charid, effectid, icon, power, tick, duration, subid, subpower, tier, flags, timestamp) VALUES(1701,310,310,3,0,229,17201035,3,288,131368,1657471549); (SqlConnection::QueryStr:364)
[07/10/22 13:23:35:304][map][error] Query: INSERT INTO char_effects (charid, effectid, icon, power, tick, duration, subid, subpower, tier, flags, timestamp) VALUES(918,324,324,1,0,169,17631107,1,369,131368,1657473684); (SqlConnection::QueryStr:364)
[07/10/22 13:23:35:304][map][error] Query: INSERT INTO char_effects (charid, effectid, icon, power, tick, duration, subid, subpower, tier, flags, timestamp) VALUES(918,314,314,2,0,172,17631107,1,30,131368,1657473687); (SqlConnection::QueryStr:364)
[07/11/22 16:39:39:250][map][error] Query: INSERT INTO char_effects (charid, effectid, icon, power, tick, duration, subid, subpower, tier, flags, timestamp) VALUES(968,310,310,2,0,90,17586049,1,288,131368,1657571770); (SqlConnection::QueryStr:364)
[07/11/22 16:39:39:250][map][error] Query: INSERT INTO char_effects (charid, effectid, icon, power, tick, duration, subid, subpower, tier, flags, timestamp) VALUES(968,326,326,11,0,145,17586049,3,382,131368,1657571825); (SqlConnection::QueryStr:364)
[07/11/22 18:48:09:717][map][error] Query: INSERT INTO char_effects (charid, effectid, icon, power, tick, duration, subid, subpower, tier, flags, timestamp) VALUES(1867,310,310,3,0,170,17102723,3,288,131368,1657579560); (SqlConnection::QueryStr:364)
[07/11/22 22:40:39:521][map][error] Query: INSERT INTO char_effects (charid, effectid, icon, power, tick, duration, subid, subpower, tier, flags, timestamp) VALUES(401,324,324,3,0,278,16922497,5,369,131368,1657593618); (SqlConnection::QueryStr:364)
[07/11/22 22:40:39:522][map][error] Query: INSERT INTO char_effects (charid, effectid, icon, power, tick, duration, subid, subpower, tier, flags, timestamp) VALUES(401,314,314,2,0,281,16922497,1,30,131368,1657593621); (SqlConnection::QueryStr:364)
[07/11/22 22:40:44:527][map][error] Query: INSERT INTO char_effects (charid, effectid, icon, power, tick, duration, subid, subpower, tier, flags, timestamp) VALUES(401,324,324,3,0,273,16922497,5,369,131368,1657593618); (SqlConnection::QueryStr:364)
[07/11/22 22:40:44:527][map][error] Query: INSERT INTO char_effects (charid, effectid, icon, power, tick, duration, subid, subpower, tier, flags, timestamp) VALUES(401,314,314,2,0,276,16922497,1,30,131368,1657593621); (SqlConnection::QueryStr:364)
[07/11/22 23:30:20:578][map][error] Query: INSERT INTO char_effects (charid, effectid, icon, power, tick, duration, subid, subpower, tier, flags, timestamp) VALUES(1708,321,321,10,0,185,17102723,3,73,131368,1657596505); (SqlConnection::QueryStr:364)
[07/11/22 23:30:20:578][map][error] Query: INSERT INTO char_effects (charid, effectid, icon, power, tick, duration, subid, subpower, tier, flags, timestamp) VALUES(1708,310,310,5,0,187,17102723,6,288,131368,1657596508); (SqlConnection::QueryStr:364)
```

Actual error
```[07/08/22 13:39:18:908][map][error] mysql_real_query: SQL_ERROR: Out of range value for column 'subid' at row 1 (1264) (SqlConnection::QueryStr:365)```

## Expected behavior

Players use effects without error

**Comment by zach2good** [comment]
Looking at the information, these all look to be related to corsair rolls:
```
INSERT INTO `status_effects` VALUES (310,'fighters_roll',131368,0,0,0,0,0,0,0,2000);
INSERT INTO `status_effects` VALUES (311,'monks_roll',131368,0,0,0,0,0,0,0,2000);
INSERT INTO `status_effects` VALUES (312,'healers_roll',131368,0,0,0,0,0,0,0,2000);
INSERT INTO `status_effects` VALUES (313,'wizards_roll',131368,0,0,0,0,0,0,0,2000);
INSERT INTO `status_effects` VALUES (314,'warlocks_roll',131368,0,0,0,0,0,0,0,2000);
INSERT INTO `status_effects` VALUES (315,'rogues_roll',131368,0,0,0,0,0,0,0,2000);
INSERT INTO `status_effects` VALUES (316,'gallants_roll',131368,0,0,0,0,0,0,0,2000);
INSERT INTO `status_effects` VALUES (317,'chaos_roll',131368,0,0,0,0,0,0,0,2000);
INSERT INTO `status_effects` VALUES (318,'beast_roll',131368,0,0,0,0,0,0,0,2000);
INSERT INTO `status_effects` VALUES (319,'choral_roll',131368,0,0,0,0,0,0,0,2000);
INSERT INTO `status_effects` VALUES (320,'hunters_roll',131368,0,0,0,0,0,0,0,2000);
INSERT INTO `status_effects` VALUES (321,'samurai_roll',131368,0,0,0,0,0,0,0,2000);
INSERT INTO `status_effects` VALUES (322,'ninja_roll',131368,0,0,0,0,0,0,0,2000);
INSERT INTO `status_effects` VALUES (323,'drachen_roll',131368,0,0,0,0,0,0,0,2000);
INSERT INTO `status_effects` VALUES (324,'evokers_roll',131368,0,0,0,0,0,0,0,2000);
```

I have _no idea_ how corsair works, but it might be related to https://github.com/LandSandBoat/server/pull/1984 if it's only started happening since that PR?

**Comment by zach2good** [comment]
A set of small, local repro steps would be immensely helpful for this sort of thing

**Comment by CatsEyeXI** [comment]
> A set of small, local repro steps would be immensely helpful for this sort of thing



> A set of small, local repro steps would be immensely helpful for this sort of thing

I'll see what I can do, no players reported this, meaning they probably don't realize that it's not working.. I only happened to catch it in the log...  I'm working on-site today so this will have to wait til i get home.

**Comment by CatsEyeXI** [comment]
and yes, we are updated beyond #1984

**Comment by WinterSolstice8** [comment]
The fix in #2272 is a good change because now the DB will match the C++ code for how wide of an integer it should store. But how we're currently storing how the phantom roll is "owned" in the subid is _painfully_ wrong... which directly relates to #845.

Status Effects aren't "owned" by anyone, so the player ID for ownership is stored in the subid of COR roll's effect so doubleup won't be buggy and boost other CORs rolls.
